pkgs/build-support/cc-wrapper/ld-wrapper.sh

```diff
@@ -135,10 +141,10 @@ if [ "$NIX_DONT_SET_RPATH" != 1 ]; then
     # so, add the directory to the rpath.
     # It's important to add the rpath in the order of -L..., so
     # the link time chosen objects will be those of runtime linking.
-    for i in ${libDirs[@]}; do
-        for j in ${libs[@]}; do
-            if [ -f "$i/lib$j.so" ]; then
-                addToRPath $i
+    for dir in "${libDirs[@]}"; do
+        for file in "$dir"/lib*.so; do
+            if [ "${libs[${file##*/}]}" ]; then
```

Comment:
What is not needed anymore? I take it the idea here is to establish a `basename => dir` mapping, whereas the generated items will contain the dir so you need to take the basename (aka `${file##*/}`).